pyuvdata/uvdata/uvdata.py

```diff
@@ -2360,6 +2380,167 @@ def _check_freq_spacing(self, raise_errors=True):
+        the size of data-like parameters by up to a factor of Nspws (the true value
+        depends on the number of unique entiries in `flex_spw_polarization_array`).
+        """
+        if self.flex_spw_polarization_array is None:
+            # There isn't anything to do, so just move along
+            return
+
+        self.polarization_array = np.unique(self.flex_spw_polarization_array)
+        self.Npols = len(self.polarization_array)
+
+        # If we have metadata only, or there was only one pol we were working with,
+        # then we do not need to do anything further aside from removing the array
+        # associated with flex_spw_polarization_array
+        if self.metadata_only or (self.Npols == 1):
+            self.flex_spw_polarization_array = None
+            return
+
+        # Otherwise, move through all of the data params
+        for name, param in zip(self._data_params, self.data_like_parameters):
+            # We need to construct arrays with the appropriate shape
+            new_shape = [self.Nblts, 1, self.Nfreqs, self.Npols]
+
+            # Pop the spw-axis if using future array shapes
+            if self.future_array_shapes:
+                del new_shape[1]
+
+            # Use fill here, since we want zeros if we are working with nsample_array
+            # or data_array, otherwise we want True if working w/ flag_array.
+            new_param = np.full(new_shape, name == "flag_array", dtype=param.dtype)
+
+            # Now we have to iterate through each spectral window
+            for spw, pol in zip(self.spw_array, self.flex_spw_polarization_array):
+                pol_idx = np.intersect1d(
+                    pol, self.polarization_array, return_indices=True
+                )[2][0]
+                spw_screen = self.flex_spw_id_array == spw
+
+                # Note that this works because pol_idx is an integer, ergo a simple
+                # slice (wherease spw_screen is a complex slice, which we can only have
+                # one of for an array).
+                if self.future_array_shapes:
+                    new_param[:, spw_screen, pol_idx] = param[:, spw_screen, 0]
+                else:
+                    new_param[:, :, spw_screen, pol_idx] = param[:, :, spw_screen, 0]
+
+            # With the new array defined and filled, set the attribute equal to it
+            setattr(self, name, new_param)
+
+        # Finally, remove the flex-pol attribute
+        self.flex_spw_polarization_array = None
+
+    def _make_flex_pol(self, raise_error=False, raise_warning=True):
+        """
+        Convert a regular UVData object into one with flex-polarization enabled.
+
+        This is an internal helper function, which is not designed to be called by
+        users, but rather individual read/write functions for the UVData object.
+        This will convert a regulat UVData object into one that uses flexible
```

Comment:
possible typo regulat?

Reply:
Well that's not right. Fixed in 8f8a8af4.